### PR TITLE
Support kovan blocks (no nonce field)

### PR DESCRIPTION
**Pull request by XertroV**
Little hack to replace `nonce` with `author` when absent (particular to Kovan blocks)

note: added simple-json dependency because it can read JSON to records whereas I couldn't see a way to do this with `decode`

### CODE REVIEW

File: src/Network/Ethereum/Web3/Types/BigNumber.purs [hunk 1]
```diff
@@ -13,10 +13,11 @@ module Network.Ethereum.Web3.Types.BigNumber
 import Prelude
 
 import Data.Foreign (Foreign)
-import Data.Foreign.Class (class Decode, class Encode, encode)
+import Data.Foreign.Class (class Decode, class Encode, decode, encode)
 import Data.Int (Radix, binary, decimal, hexadecimal, floor) as Int
 import Data.Maybe (Maybe(..))
 import Data.Module (class LeftModule, class RightModule)
+import Simple.JSON (class ReadForeign)
```
Comment (blinky3713) [inline src/Network/Ethereum/Web3/Types/BigNumber.purs]: Doesn't look like you're actually using this package other than to implement the class, is there a reason for this?

**Comment by XertroV** [comment]
Link to #72

File: src/Network/Ethereum/Web3/Types/BigNumber.purs (hunk 1 above)
Comment (XertroV) [reply src/Network/Ethereum/Web3/Types/BigNumber.purs]: So in `BigNumber.purs` we just need to implement the class, but I use the `read` function here: https://github.com/f-o-a-m/purescript-web3/pull/74/files/99c67712059920800b97b86c4ad26e571e6da17e#diff-ededbfc84072116829037439fdc1c254R285 which requires all types `Block` depends on to implement the `ReadForeign` class.

Comment (XertroV) [reply src/Network/Ethereum/Web3/Types/BigNumber.purs]: PS. If you haven't played with simple-json it's super nice (uses stuff like `RowCons` and `RowToList` and that good stuff to natively handle records, which I think is much nicer than the usual `Generics` method of JSON conversion)

**Comment by blinky3713** [comment]
thanks, looks good

**Pull request merged**